react-components/src/tools/ManageTools.js

```diff
@@ -0,0 +1,524 @@
+            page
+        );
+    };
+
+    const onToolFilterValueChanged = (event) => {
+        let updatedFilter = event.target.value;
+        setToolFilterValue(updatedFilter);
+        refreshListing(updatedFilter, "", order, orderBy, rowsPerPage, 0);
+    };
+
+    const handleSearch = (searchTerm) => {
+        setToolFilterValue("");
+        refreshListing("", searchTerm, order, orderBy, rowsPerPage, 0);
+    };
+
+    const onRequestSort = (event, property) => {
+        const isDesc = orderBy === property && order === "desc";
+        refreshListing(
+            toolFilterValue,
+            searchTerm,
+            isDesc ? "asc" : "desc",
+            property,
+            rowsPerPage,
+            page
+        );
+    };
+
+    const onChangePage = (event, page) => {
+        refreshListing(
+            toolFilterValue,
+            searchTerm,
+            order,
+            orderBy,
+            rowsPerPage,
+            page
+        );
+    };
+
+    const onChangeRowsPerPage = (event) => {
+        refreshListing(
+            toolFilterValue,
+            searchTerm,
+            order,
+            orderBy,
+            event.target.value,
+            0
+        );
+    };
+
+    const onToolSelection = (tool) => {
+        if (selectedTool === tool) {
+            tool = null;
+        }
+        setSelectedTool(tool);
+        setNumToolsSelected(tool ? 1 : 0);
+        presenter.onToolSelectionChanged(tool);
+    };
+
+    const baseId = build(parentId, ids.MANAGE_TOOLS.VIEW);
+
+    return (
+        <Fragment>
+            <StyledToolbar
+                parentId={baseId}
+                presenter={presenter}
+                intl={intl}
+                searchTerm={searchTerm}
+                selectedTool={selectedTool}
+                toolFilterValue={toolFilterValue}
+                onRefreshClicked={onRefreshClicked}
+                onToolFilterValueChanged={onToolFilterValueChanged}
+                handleSearch={handleSearch}
+            />
+            <div className={classes.container}>
+                <LoadingMask loading={loading}>
+                    <Table size="small">
+                        <ToolListing
+                            parentId={baseId}
+                            toolList={toolList}
+                            presenter={presenter}
+                            selectedTool={selectedTool}
+                            onToolSelection={onToolSelection}
+                        />
+                        <EnhancedTableHead
+                            selectable={true}
+                            numSelected={numToolsSelected}
+                            rowCount={toolList ? toolList.tools.length : 0}
+                            baseId={parentId}
+                            columnData={TABLE_COLUMNS}
+                            order={order}
+                            orderBy={orderBy}
+                            onSelectAllClick={() => onToolSelection(null)}
```

Comment:
Not sure if am understand this right. What happens when the checkbox in the header is selected ? Since you are passing in `null` it reset selection all the time ? When nothing is selected, this is a no-op. It can get confusing for users.

Reply:
That's a good point - you're right, nothing will happen if they click Select All when nothing is already selected.  I added it mostly because I thought having the checkboxes was a better indicator to the user which row was actually selected (right now it's a bit faint, I think). Maybe for now I'll remove the checkboxes and in another PR we can look into changing the theme so highlighted rows stand out more?

Reply:
I am ok with removing checkbox for now.